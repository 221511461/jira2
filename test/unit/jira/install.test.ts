/* eslint-disable @typescript-eslint/no-explicit-any */
import testTracking from '../../setup/tracking';

describe('Webhook: /events/installed', () => {
  let installation;
  let body;
  let install;

  beforeEach(async () => {
    body = {
      baseUrl: 'https://test-host.jira.com',
      clientKey: 'abc123',
      sharedSecret: 'ghi345',
    };

    installation = {
      id: 19,
      jiraHost: body.baseUrl,
      clientKey: body.clientKey,
      enabled: true,
      secrets: 'def234',
      sharedSecret: body.sharedSecret,
      subscriptions: jest.fn().mockResolvedValue([]),
    };

    td.when(models.Installation.install(td.matchers.anything()))
      // Allows us to modify installation before it's finally called
      .thenDo(async () => installation);

    install = (await import('../../../src/jira/install')).default;
  });

  it('Install', async () => {
    await testTracking();
    const req = { log: { info: jest.fn() }, body };
    const res = { sendStatus: jest.fn(), on: jest.fn() };

    await install(req as any, res as any);
    expect(res.sendStatus).toHaveBeenCalledWith(204);
  });
});
