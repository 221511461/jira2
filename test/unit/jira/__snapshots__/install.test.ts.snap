// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webhook: /events/installed Install 1`] = `"jira.v0.Action"`;

exports[`Webhook: /events/installed Install 2`] = `
Object {
  "actionSource": "webhook",
  "association": "installation",
  "type": "created",
}
`;
